### PR TITLE
Switched access token for id token during logout flow

### PR DESCRIPTION
**Pull request by StevenPlettinxKlarrio**
We were having issue because we were passing the access token during logout when the id token was required

### CODE REVIEW

**Review by TomBillietKlarrio — COMMENTED** [review]
good fix but wondering, no tests for this?

File: main.go [hunk 1]
```diff
@@ -95,13 +95,22 @@ func handler(w http.ResponseWriter, r *http.Request) {
...
+
 	if fw.LogoutPath != "" && uri.Path == fw.LogoutPath {
 		logger.WithFields(logrus.Fields{
 			"logout-path":      fw.LogoutPath,
 			"post-redirect-to": fw.PostLogoutPath,
 		}).Info("handling logout ")
 		endSession(sessionKey, w, r)
-		if logoutError := oidcApi.LogOut(rawAccessToken); logoutError != nil {
+		if logoutError := oidcApi.LogOut(*idToken);
+		logoutError != nil {
```
Comment (gert-verbruggen-klarrio) [inline main.go:113]: I find this more readable on one line.

File: session/inventory.go [hunk 2]
```diff
@@ -72,6 +72,24 @@ func (i *Inventory) sessionToken(sessionKey string) (availableToken *oidc.Token,
...
 }
 
+func (i *Inventory) GetIdToken(sessionKey string) (idToken *string, idTokenExists bool) {
+	sessionState, sessionExists := i.stateMap.Get(sessionKey)
+	if sessionExists {
+		switch val := sessionState.Value().(type) {
+		case *oidc.Token:
+			return &val.IDToken, true
+		case *TokenPromise:
+			return nil, false
```
Comment (gert-verbruggen-klarrio) [inline session/inventory.go:82]: This case is the same as `default`

**Review by gert-verbruggen-klarrio — APPROVED** [review]

**Comment by gert-verbruggen-klarrio** [comment]
Is it straightforward to add a test case in this forked project?

**Comment by StevenPlettinxKlarrio** [comment]
> Is it straightforward to add a test case in this forked project?

apparently, it is. this project is so heavily modified it's unrecognizable, and most of the tests we have are written by us anyway. I added one to test the post redirect URL and the retrieval of the id token from the session map.

**Review by TomBillietKlarrio — APPROVED** [review]

**Pull request merged**